src/gennylib/src/PoolFactory.cpp

```diff
@@ -240,15 +240,18 @@ mongocxx::options::pool PoolFactory::makeOptions() const {
 std::unique_ptr<mongocxx::pool> PoolFactory::makePool() const {
     auto uriStr = makeUri();
 
+    std::string username{*_config->get(OptionType::kAccessOption, "Username")};
     std::string password{*_config->get(OptionType::kAccessOption, "Password")};
-
-    auto redactedUriStr = std::regex_replace(uriStr, std::regex(password), "[REDACTED]" );
+    auto credentials = username + ':' + password + '@';
+    auto redactedCredentials = username + ':' + "[REDACTED]" + '@';
+    auto credPosition = uriStr.find(credentials);
+    auto redactedUriStr = credPosition == std::string::npos
+                        ? uriStr
+                        : uriStr.replace(credPosition, credentials.length(), redactedCredentials);
```

Comment:
@thessem Do you think it would be worth it to extract these lines into a helper method (like `std::string PoolFactory::getRedactedUri()`), which should be more testable, and add test for this method?